### PR TITLE
Implemente Component api

### PR DESCRIPTION
**Pull request by zzxwill**
- Update app list/show, implmented component show
- Update trait attach/detach with the introduction of component
- implemented component delete API

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]

File: pkg/oam/common.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core"
+	certmanager "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/api/types"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
```
Comment (wonderflow) [inline pkg/oam/common.go:15]: nit format

File: pkg/server/apis/types.go [hunk 2]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
-	EnvName      string       `json:"env_name"`
-	Name         string       `json:"name"`
-	Flags        []CommonFlag `json:"flags"`
-	WorkloadName string       `json:"workload_name"`
-	AppGroup     string       `json:"app_group,omitempty"`
-	Staging      string       `json:"staging,omitempty"`
+	EnvName       string       `json:"env_name"`
+	Name          string       `json:"name"`
+	Flags         []CommonFlag `json:"flags"`
+	ComponentName string       `json:"component_name"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:66]: ApiChanged, will dashboard frontend affected?

File: pkg/server/apis/types.go [hunk 3]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
+	AppGroup      string       `json:"app_group,omitempty"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:67]:
```suggestion
	AppGroup      string       `json:"appGroup,omitempty"`
```

We MUST align our style, it feels really bad to see we have so much naming problem.

File: pkg/server/apis/types.go [hunk 4]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
+	Staging       string       `json:"staging,omitempty"`
 }
 
-type ApplicationStatusMeta struct {
+type ComponentMeta struct {
+	Name     string                        `json:"name"`
 	Status   string                        `json:"Status,omitempty"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:73]:
```suggestion
	Status   string                        `json:"status,omitempty"`
```

File: pkg/server/apis/types.go [hunk 5]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
-	Workload corev1alpha2.ComponentSpec    `json:"Workload,omitempty"`
+	Workload runtime.RawExtension          `json:"workload,omitempty"`
 	Traits   []corev1alpha2.ComponentTrait `json:"Traits,omitempty"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:75]:
```suggestion
	Traits   []corev1alpha2.ComponentTrait `json:"traits,omitempty"`
```

File: pkg/server/apis/types.go [hunk 6]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
 }
 
+type ApplicationMeta struct {
+	Status     string          `json:"Status,omitempty"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:79]:
```suggestion
	Status     string          `json:"status,omitempty"`
```

File: pkg/server/apis/types.go [hunk 7]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
...
+	Components []ComponentMeta `json:"Components,omitempty"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:80]:
```suggestion
	Components []ComponentMeta `json:"components,omitempty"`
```

File: pkg/server/apis/types.go (hunk 2 above)
Comment (wonderflow) [inline pkg/server/apis/types.go:66]:
```suggestion
	ComponentName string       `json:"componentName"`
```

File: pkg/server/apis/types.go [hunk 8]
```diff
@@ -60,20 +60,26 @@ type TraitMeta struct {
 
 //used to present trait which is to be attached and, of which parameters are set
 type TraitBody struct {
-	EnvName      string       `json:"env_name"`
-	Name         string       `json:"name"`
-	Flags        []CommonFlag `json:"flags"`
-	WorkloadName string       `json:"workload_name"`
-	AppGroup     string       `json:"app_group,omitempty"`
-	Staging      string       `json:"staging,omitempty"`
+	EnvName       string       `json:"env_name"`
```
Comment (wonderflow) [inline pkg/server/apis/types.go:63]:
```suggestion
	EnvName       string       `json:"envName"`
```

**Review by wonderflow — CHANGES_REQUESTED** [review]

File: pkg/server/apis/types.go (hunk 2 above)
Comment (zzxwill) [reply pkg/server/apis/types.go:66]: Yes, I will sync with the front-end developer and update the api documentation.

**Comment by zzxwill** [comment]
@wonderflow Sorry all your commits logs lost as I forced push the commits.

File: pkg/oam/application.go [hunk 9]
```diff
@@ -79,17 +79,17 @@ func ListComponents(ctx context.Context, c client.Client, opt Option) ([]Compone
...
-				Name:        com.ComponentName,
-				App:         a.Name,
-				Workload:    workload,
-				Status:      types.StatusDeployed,
-				Traits:      traitAlias,
+				// Name: com.ComponentName,
+				App: a.Name,
+				//Workload:    workload,
+				Status: types.StatusDeployed,
+				//Traits:      traitAlias,
```
Comment (wonderflow) [inline pkg/oam/application.go:92]: If we don't need any more, please delete them

**Review by wonderflow — APPROVED** [review]

**Pull request merged**